src/DearImGui/Vulkan/Types.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module DearImGui.Vulkan.Types
+  ( addVulkanTypesTable )
+  where
+
+-- containers
+import qualified Data.Map.Strict as Map
+  ( fromList )
+
+-- inline-c
+import qualified Language.C.Inline.Context as C
+import qualified Language.C.Types          as C
+
+-- vulkan
+import qualified Vulkan
+
+vulkanTypesTable :: C.TypesTable
+vulkanTypesTable = Map.fromList
+  [ ( C.TypeName "VkAllocationCallbacks", [t|Vulkan.AllocationCallbacks|] )
+  , ( C.TypeName "VkDescriptorPool"     , [t|Vulkan.DescriptorPool|] )
+  , ( C.TypeName "VkCommandBuffer_T"    , [t|Vulkan.CommandBuffer_T|])
+  , ( C.TypeName "VkDevice_T"           , [t|Vulkan.Device_T|])
+  , ( C.TypeName "VkInstance_T"         , [t|Vulkan.Instance_T|])
+  , ( C.TypeName "VkPhysicalDevice_T"   , [t|Vulkan.PhysicalDevice_T|])
+  , ( C.TypeName "VkPipeline"           , [t|Vulkan.Pipeline|] )
+  , ( C.TypeName "VkPipelineCache"      , [t|Vulkan.PipelineCache|] )
+  , ( C.TypeName "VkQueue_T"            , [t|Vulkan.Queue_T|])
+  , ( C.TypeName "VkRenderPass"         , [t|Vulkan.RenderPass|] )
+  , ( C.TypeName "VkResult"             , [t|Vulkan.Result|] )
+  , ( C.TypeName "VkSampleCountFlagBits", [t|Vulkan.SampleCountFlagBits|] )
+  ]
+
+addVulkanTypesTable :: C.Context -> C.Context
+addVulkanTypesTable ctx@( C.Context { C.ctxTypesTable } ) = ctx { C.ctxTypesTable = ctxTypesTable <> vulkanTypesTable }
```

Comment:
I think what you usually do with `inline-c` is to just provide a `C.Context`, and then you can use its `Monoid` instance to combine multiple context together. So this would be:

```suggestion
vulkanContext :: C.Context
vulkanContext = mempty { C.ctxTypesTable = vulkanTypesTable }
```